stages/org.osbuild.nm.conn

```diff
@@ -0,0 +1,202 @@
+    "required": ["id","uuid","type"],
+    "type": "object",
+    "properties": {
+      "uuid": {
+        "type": "string"
+      },
+      "id": {
+        "type": "string"
+      },
+      "type": {
+        "type": "string"
+      },
+      "autoconnect": {
+        "type": "boolean"
+      },
+      "interface-name": {
+        "type": "string"
+      },
+      "autoconnect-priority": {
+        "type": "number"
+      }
+    }
+  },
+  "ethernet": {
+    "additionalProperties": false,
+    "required": ["connection"],
+    "type": "object",
+    "properties": {
+      "connection": {
+        "allOf": [
+          {"$ref": "#/definitions/connection"},
+          {"$ref": "#/definitions/802-3-ethernet-connection"}
+        ]
+      },
+      "ipv4": {
+        "$ref": "#/definitions/ipv4"
+      },
+      "ipv6": {
+        "$ref": "#/definitions/ipv6"
+      },
+      "ethernet": {
+        "$ref": "#/definitions/802-3-ethernet"
+      }
+    }
+  },
+  "802-3-ethernet-connection": {
+    "additionalProperties": true,
+    "required": ["type"],
+    "type": "object",
+    "properties": {
+      "type": {
+        "enum": ["802-3-ethernet","ethernet"]
+      }
+    }
+  },
+  "ipv4": {
+    "oneOf": [
+      {"$ref": "#/definitions/ipv4-ignore"},
+      {"$ref": "#/definitions/ipv4-auto"},
+      {"$ref": "#/definitions/ipv4-manual"}
+    ]
+  },
+  "ipv4-ignore": {
+    "additionalProperties": false,
+    "required": ["method"],
+    "properties": {
+      "method": {
+        "enum": ["ignore"]
+      }
+    }
+  },
+  "ipv4-auto": {
+    "additionalProperties": false,
+    "required": ["method"],
+    "properties": {
+      "method": {
+        "enum": ["auto"]
+      },
+      "dns-search": {
+        "type": "string"
+      }
+    }
+  },
+  "ipv4-manual": {
+    "additionalProperties": false,
+    "required": ["method"],
+    "properties": {
+      "method": {
+        "enum": ["manual"]
+      },
+      "address": {
+        "type": "array",
+        "items": {
+          "$ref": "#/definitions/ipv4-address"
+        }
+      }
+    }
+  },
+  "ipv4-address": {
+    "type": "string"
+  },
+  "ipv6": {
+    "oneOf": [{"$ref": "#/definitions/ipv6-ignore"}]
+  },
+  "ipv6-ignore": {
+    "additionalProperties": false,
+    "required": ["method"],
+    "properties": {
+      "method": {
+        "enum": ["ignore"]
+      }
+    }
+  },
+  "802-3-ethernet": {
+    "additionalProperties": true,
+    "type": "object",
+    "properties": {
+      "mac-address": {
+        "type": "string"
+      }
+    }
+  }
+},
+"additionalProperties": false,
+"required": ["filename", "config"],
+"properties": {
+  "filename": {
+    "type": "string",
+    "pattern": "^[\\w.-]{1,255}$"
+  },
+  "config": {
+    "oneOf": [{"$ref": "#/definitions/ethernet"}]
+  }
+}
+"""
+
+
+USR_PATH = "usr/lib/NetworkManager/system-connections"
+
+
+def main(tree, options):
+    filename = options["filename"]
+    config = options["config"]
```

Comment:
I think `config`, `cfg`, and later `opts` are a bit confusing, especially that `config` is kind of the input and `cfg` is the output, I'll try to think of better names.

Reply:
Maybe `cfgfile` for the actual. Could also have `settings` in the actual schema instead of config, since it is `nm-settings(5)`